R/ggplotMDS.R

```diff
@@ -96,105 +139,209 @@ ggplotMDS <- function(DGEdata,
+            if (!missing(symShape) & any(length(symShape) != 1, !.is_valid_symbolShapes_cxplot(symShape))) {
+                warning("symShape must be a singular value of class 'character'. Assigning default value 'circle'.")
+                symShape <- "circle"
+            }
+            shapes <- symShape
+        } else {
+            shapes <- .get_valid_symbolShapes_cxplot()[1:8]
+        }
+
+        intercept_flag <- FALSE
+        if (!missing(hlineIntercept)) {
+            if (!is.null(hlineIntercept)) {
+                intercept_flag <- TRUE
+                if (!(is.numeric(hlineIntercept) & length(hlineIntercept) == 1)) {
+                    warning("hlineIntercept must be a singular numeric value. Ignoring hlineIntercept.")
+                    hlineIntercept <- NULL
+                }
+            }
+        }
+        if (!missing(vlineIntercept)) {
+            if (!is.null(vlineIntercept)) {
+                intercept_flag <- TRUE
+                if (!(is.numeric(vlineIntercept) & length(vlineIntercept) == 1)) {
+                    warning("vlineIntercept must be a singular numeric value. Ignoring vlineIntercept.")
+                    vlineIntercept <- NULL
+                }
+            }
+        }
+
+        if (intercept_flag) {
+
+            if (!missing(reflineColor)) {
+                if (any(is.null(reflineColor), !is.character(reflineColor), length(reflineColor) != 1)) {
+                    warning("reflineColor must be a singular value of class character and must specify the name of the color or the rgb value. Assigning default value 'red'.")
+                    reflineColor <- "red"
+                } else if (.rgbaConversion(reflineColor) == "invalid value") {
+                    warning("Color specified is not valid. Assigning default value 'red'.")
+                    reflineColor <- "red"
+                }
+            }
+
+            if (!missing(reflineSize)) {
+                if (any(is.null(reflineSize), !is.numeric(reflineSize), length(reflineSize) != 1, reflineSize < 0)) {
+                    warning("reflineSize must be a singular numeric value greater than 0. Assigning default value '0.5'.")
+                    reflineSize <- 0.5
+                }
+            }
+        }
     }
-    if (!missing(sizeBy)) {
-        assertthat::assert_that(length(sizeBy) == ncol(DGEdata),
-                                msg = "sizeBy should be the length of the number of columns in DGEdata.")
+
+    if (plotType == "ggplot") {
+
+        if (!missing(labels)) {
+            if (any(!is.numeric(labelSize), length(labelSize) != 1, labelSize < 0)) {
+                warning("labelSize should be singular numeric value and greater than zero. Assigning default value 3.")
+                labelSize <- 3
+            }
+        }
+
+        intercept_flag <- FALSE
+        intercept_length <- 0
+        if (!missing(hlineIntercept)) {
+            if (!is.null(hlineIntercept)) {
```

Comment:
as there is no else for the second if, code can be reduced by combining both ifs with **&&**